### PR TITLE
replaces jsonpath-plus module

### PR DESCRIPTION
**Pull request by jaredperreault-okta**
`jsonpath-plus` in vulnerable to RCE: https://security.snyk.io/vuln/SNYK-JS-JSONPATHPLUS-7945884

### CODE REVIEW

File: test/spec/util/jsonpath.ts [hunk 1]
```diff
@@ -1,14 +1,98 @@
...
+    })).toEqual('pass');
+
+    expect(jsonpath({
+      path: 'foo.bar',
+      json: { foo: { bar: { baz: 'pass' } } }
+    })).toEqual({ baz: 'pass' });
+
+    expect(jsonpath({
+      path: '$.foo.bar[12]',
+      json: { foo: { bar: Array(15).fill('pass') } }
```
Comment (shawyu-okta) [inline test/spec/util/jsonpath.ts:18]: Can we update array access tests like this to be more like
```
const values = Array(n).fill('fail');
values[12] = 'pass';

// assertions
expect(jsonpath({...});
```

File: yarn.lock [hunk 2]
```diff
@@ -3077,16 +3077,16 @@ ansi-red@^0.1.1:
   dependencies:
     ansi-wrap "0.1.0"
 
-ansi-regex@^2.0.0, ansi-regex@^4.1.0, ansi-regex@^4.1.1:
-  version "4.1.1"
-  resolved "https://registry.yarnpkg.com/ansi-regex/-/ansi-regex-4.1.1.tgz#164daac87ab2d6f6db3a29875e2d1766582dabed"
-  integrity sha512-ILlv4k/3f6vfQ4OoP2AGvirOktlQ98ZEL1k9FaQjxa3L1abBgbuTDAdPOpvbGncC0BTVQrl+OM8xZGK6tWXt7g==
-
-ansi-regex@^3.0.1, ansi-regex@^6.0.1:
```
Comment (shawyu-okta) [inline yarn.lock:3085]: I was expecting to see `jsonpath-plus` removed in this diff but it isn't

Comment (jaredperreault-okta) [reply yarn.lock:3085]:
```bash
jared:auth-js$ yarn why jsonpath-plus
yarn why v1.22.19
[1/4] 🤔  Why do we have the module "jsonpath-plus"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "jsonpath-plus@6.0.1"
info Reasons this module exists
   - "_project_#@okta#samples.express-embedded-sign-in-widget#@okta#okta-signin-widget#@okta#okta-auth-js" depends on it
   - Hoisted from "_project_#@okta#samples.express-embedded-sign-in-widget#@okta#okta-signin-widget#@okta#okta-auth-js#jsonpath-plus"
   - Hoisted from "_project_#@okta#samples.react-embedded-auth-with-sdk#@okta#okta-signin-widget#@okta#okta-auth-js#jsonpath-plus"
   - Hoisted from "_project_#@okta#samples.webpack-spa#@okta#okta-signin-widget#@okta#okta-auth-js#jsonpath-plus"
info Disk size without dependencies: "372KB"
info Disk size with unique dependencies: "372KB"
info Disk size with transitive dependencies: "372KB"
info Number of shared dependencies: 0
✨  Done in 0.56s.
```

Comment (jaredperreault-okta) [reply yarn.lock:3085]: We include samples in the repo

File: lib/util/jsonpath.ts [hunk 3]
```diff
@@ -1,8 +1,33 @@
...
+/* eslint complexity:[0,8] */
+export function jsonpath({ path, json }) {
+  const steps: string[] = [];
+  let match: RegExpExecArray | null;
+	while ((match = jsonpathRegex.exec(path)) !== null) {
+		const step = match?.groups?.step ?? match?.groups?.index;
+    if (step) {
+      steps.push(step);
+    }
+	}
```
Comment (shawyu-okta) [inline lib/util/jsonpath.ts:12]: Formatting is kind of wonky, it's not failing lint?

**Review by shawyu-okta — APPROVED** [review]
just a comment about the indentation/whitespace of the main util file but overall lgtm

**Review by shawyu-okta — APPROVED** [review]

**Pull request merged**